src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -145,6 +145,21 @@ public bool TryTakeFirst(out TValue first)
             return TryRemove(GetKey(sortedValues.Min), out first);
         }
         
+        /// <summary>
+        /// Returns best element of each bucket in supplied comparer order.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.Synchronized)]
+        public IEnumerable<TValue> GetFirsts()
+        {
+            SortedSet<TValue> sortedValues = new(_sortedComparer);
+            foreach (KeyValuePair<TGroupKey, SortedSet<TValue>> bucket in _buckets)
+            {
+                sortedValues.Add(bucket.Value.Max!);
```

Comment:
In TryTakeFirst we are using Min, and here we are using Max? 

If this is a mistake and both should be Min, TryTakeFirst can use this method as base.

Reply:
This is a mistake - good catch. I will fix it and reuse